### ISSUE TITLE
Is the EEG encoder only one-layer LSTM?

### ISSUE THREAD

**Issue by ChiShengChen** [prajwalsingh/EEGStyleGAN-ADA#12]
Thanks for the interesting work!
I have some problems about the EEG encoder in `./EEGClip/EEG_encoder.py`, 
In your paper, you mentioned that the EEG encoder is the multilayer LSTM network, but how I found in `./EEGClip/EEG_encoder.py` and `config.py` the code only has single-layer LSTM, maybe I overlooked something?
![2024-03-13 02-01-05 的螢幕擷圖](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/22126443/e37df955-9270-469a-8899-f98e6d319905)
![2024-03-13 02-09-04 的螢幕擷圖](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/22126443/5f2d8de6-f6ae-4aaa-b6df-d6f816bc9e37)

**Comment by prajwalsingh** [comment]
Hi @ChiShengChen , thank you for reading our work.

You have not overlooked anything, and this is correct only. Our initial experiments used LSTM with four layers, as shown in EEG2Feat. We used the same for EEG_encoder also. Due to computation time we kept it as 1 for EEG CLIP. You can train the network with four layers also, and it will work.

**Comment by ChiShengChen** [comment]
@prajwalsingh Thanks for quick reply!  But I still bother by the code structure, is the EEG-CLIP's EEG encoder (one-layer LSTM) different from the final EEGStyleGAN-ADA's (4-layer LSTM)? To my understanding, the EEG-CLIP's EEG encoder is the same as EEGStyleGAN-ADA's, and the code `./EEGClip` only for showing  the EEG-CLIP demo but not exactly used the one-layer LSTM as EEG encoder on the paper right?
Thanks for your patient!

**Comment by prajwalsingh** [comment]
Yes, you are right. Both are the same architecture; they differ in depth. We have shown an application of EEGClip for image retrieval tasks (initially, that was the plan only). 

We haven't tried using the pre-trained EEG encoder weights using the EEGClip method for image generation.

For EEGStyleGAN-ADA, we trained the EEG encoder using triplet loss and later used it for conditional training using the GAN method.
